resources/2023-11-application-diagnostics-prototype.md

```diff
@@ -0,0 +1,215 @@
+# Title
+
+* **Status**: Pending
+* **Author**: Ryan Nowak (`@rynowak`)
+
+## Overview
+
+*This design-note describes a prototype we plan to build for application diagnostics. As this document is describing a prototype, it will be less complete than other design proposals.*
+
+Our goal is to prototype an end-to-end scenario for **application** diagnostics that emcompasses:
+
+- A platform engineer configures a metrics dashboard (Prometheus+Grafana) and a logging dashboard (probably Azure to start) and wiring these up to a Radius environment.
+- A developer deploys an application into the environment, and has metrics and log collection automatically configured for the application.
+- An SRE can configure alerts for the metrics. When they recieve an alert, they can use the metadata from the metrics to:
+  - Navigate to the application graph and understand the application architecture.
+  - Navigate to the logs and craft a query to understand the problem.
+  - *stretch* understand when and how the problem was introduced based on logs.
+
+## Terms and definitions
+
+| Term                 | Definition                                                                                                                                                                                                                                                              |
+| -------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Diagnostics          | In this context, we're referring to metrics, tracing and logs. There are other technologies that could be considered diagnostic data such as profiling and crash-dumps, but they are out of scope for this discussion.                                                  |
```

Comment:
Just to clarify, the other diagnostics sources (profiles, dumps etc) are out of scope for the demo/prototype but most certainly in scope for the overall diagnostics story?

Reply:
Yes that's what I meant 👍. This prototype is focused on the things that OTEL supports.